DX-1023: Add regex for base url

CahidArda opened the pull request:
when an invalid url is passed to the client (without https like `eu1-flying-whale-434343.upstash.io`), the client still tries to call the endpoint but gets invalild URL error. Instead of stopping there, it continues retrying with exponential backoff.

For instance, in ratelimit, there is a 5 second timeout. Since the exponential backoff takes 12 seconds in total, error is never reflected to the console/network and the user simply gets the timeout response.

With this change, client raises an error right away and ratelimit doesn't return the default timeout response.

ogzhanolguncu commented on pkg/http.ts:
```diff
@@ -125,6 +125,10 @@ export class HttpClient implements Requester {
     };
 
     this.baseUrl = config.baseUrl.replace(/\/$/, "");
+    const urlRegex = /^https?:\/\/[^\s/$.?#].[^\s]*$/;
```
Descriptive comment about what this regex does would be good. I know its somewhat clear now, but we can always make it more clearer.

CahidArda commented:
fixed CI tests in this PR aswell.

Here are the changes:
- removed `link` from cloudflare-workers/package.json to fix cloudflare-workers-local test [bdd5a66](https://github.com/upstash/redis-js/pull/1157/commits/bdd5a66b37e627048fcd478fa4090cb7529f8382)
- updated bun.lockb file to fix nodejs-local test [fc93463](https://github.com/upstash/redis-js/pull/1157/commits/fc93463e6cf035328c2fd47616b41b935f8ecb79)
- bumped node version of vercel deployed tests, which didn't actually solve the issue [7bdd2b2](https://github.com/upstash/redis-js/pull/1157/commits/7bdd2b266aa6d3c769c92a3418616984d1f556b2). Updated nodejs versions of sdk-redis and sdk-redis-edge projects at Redis

CahidArda commented on pkg/error.ts:
```diff
@@ -7,3 +7,10 @@ export class UpstashError extends Error {
+
+export class UrlError extends Error {
+  constructor(url: string) {
+    super(`Upstash Redis client was passed an invalid URL: "${url}"`);
```
updated message to include https requirement. Didn't mention ports since the client uses http
